lib/bundler/settings.rb

```diff
@@ -26,13 +27,31 @@ def [](name)
       end
     end
 
+    # In Bundler pre 2.0 any command ever set was stored in the local config
+    # file. In Bundler 2.0 they are only saved for that current command and
+    # `bundle config <setting name> <setting value>` is used for setting
+    # remembered arguments.
+    # See https://trello.com/c/yGsPNDpg
     def []=(key, value)
+      @use_current ||= (Bundler::VERSION.split(".")[0].to_i >= 2)
```

Comment:
As smlance and I talked about. The issue is that there are tons of instances where specs or lib assume auto remembered flags. So I'm going to go through and refactor them all to use the new style.